Added reference guide for sensuctl.

mercul3s opened the pull request:
Updated sensuctl install guide to match current version, and moved some
content from there into the ref guide.

calebhailey commented:
@portertech gonna need your eyeballs on this, since this has never been documented before. :) 

![noidea](https://cloud.githubusercontent.com/assets/141373/11229200/db972dee-8d47-11e5-870d-567804bb3da7.jpg)

calebhailey commented:
Added a checklist to the PR description.

portertech commented on source/docs/0.20/configuration.md:
```diff
@@ -0,0 +1,380 @@
+## Configuration Load Order {#configuration-load-order}
+
+Sensu configuration can be provided via three distinct sources: environment
+variables, a configuration file, and one or more directories containing
```
mention nested directories?

portertech commented on source/docs/0.20/configuration.md:
```diff
@@ -0,0 +1,380 @@
+}
+~~~
+
+## Configuration Load Order {#configuration-load-order}
```
Are we going to go w/ header caps?

calebhailey replied:
Fixed w/ https://github.com/heavywater/sensu-docs/commit/e4cb7e5acd35aee0511fca1cd223282152eab52c

portertech commented on source/docs/0.20/configuration.md:
```diff
@@ -0,0 +1,380 @@
+   _NOTE: Environment variables will override the configuration file (`-c`) and
+   configuration directory (`-d`) locations provided by the init script._
+
+2. Sensu loads configuration settings from the following environment variables
```
Perhaps make this #1 to help explain how init gets -c and -d :+1: (it's also the base configuration)

calebhailey replied:
But wait, do the environment variables override -c and -d? If so I think they should be mentioned second. Also, I'm adding some more content RE: runtime configuration, because it occurs to me that this is also completely undocumented.

portertech commented on source/docs/0.20/configuration.md:
```diff
@@ -0,0 +1,380 @@
+
+The following is an example Sensu configuration snippet, a JSON configuration
+file located at `/etc/sensu/config.json`. This Sensu configuration snippet
+provides Sensu with information it needs to communicate with other Sensu
```
"with the Sensu transport and data store" - no direct communication

calebhailey replied:
Fixed w/ https://github.com/heavywater/sensu-docs/commit/42289b15da0aaaae0d6579e553058d48f087af24

portertech commented:
![203245sdf0](https://cloud.githubusercontent.com/assets/149630/11252065/12e1a5f8-8de9-11e5-9d47-832ebc65ac2f.gif)

mercul3s committed:
Added reference guide for sensuctl.

Updated sensuctl install guide to match current version, and moved some
content from there into the ref guide.

palourde commented on content/sensu-core/2.0/reference/sensuctl.md:
```diff
@@ -0,0 +1,185 @@
+sensuctl can be configured to return JSON instead of the default human-readable
+format.
+
+> sensuctl check info marketing-site --format json
```
It looks like this format transforms the two consecutive hyphens `--` by an en dash `–` once generated by hugo, so when pasting and running that command:

```bash
$ sensuctl check info marketing-site –format json
[...]
Error: invalid argument(s) received
```

mercul3s replied:
Whoops, that to be wrapped in a highlight block!

palourde reviewed (approved):
Just a small problem, otherwise 🎉 🎉 🎉 

On another note, I feel like the getting started guide for sensuctl might contain too much information and the installation section should use the platform blocks. That being said, I've submitted https://github.com/sensu/sensu-docs/pull/307 which will make it easier to do that so let's have your PR merged and we can refactor the guide after!

mercul3s commented:
@palourde fixed the highlight block. Agreed on refactoring to include platform blocks and trimming it down a bit!

palourde commented:
LGTM 👍
